opal/opal_hot_reloader.rb

```diff
@@ -8,22 +10,40 @@ def connect_to_websocket(port)
     host = `window.location.host`.sub(/:\d+/, '')
     host = '127.0.0.1' if host == ''
     ws_url = "#{host}:#{port}"
-    puts "Connecting to #{ws_url}"
-    `ws = new WebSocket('ws://' + #{ws_url});`
-    `console.log(ws);
-    
-    ws.onmessage = #{lambda { |e| reload(e) }}
-`
+    puts "Hot-Reloader connecting to #{ws_url}"
+    %x{
+      ws = new WebSocket('ws://' + #{ws_url});
+      // console.log(ws);
+      ws.onmessage = #{lambda { |e| reload(e) }}
+    }
   end
-  #  // ws.onmessage = #{lambda { |e| alert(`e.data`); eval(`e.data`) }}
+
   def reload(e)
-    code = `e.data`
-    eval code
-    if @reload_post_callback
-      @reload_post_callback.call
-    else
-      puts "not reloading code"
+    reload_request = JSON.parse(`e.data`)
+    puts "Received reload request: #{reload_request.inspect}"
+    if reload_request[:type] == "ruby"
+      puts "Reloading ruby #{reload_request[:filename]}"
+      eval reload_request[:source_code]
+      if @reload_post_callback
+        @reload_post_callback.call
+      else
+        puts "not reloading code"
+      end
+    end
+    if reload_request[:type] == "css"
+      url = reload_request[:url]
+      puts "Reloading CSS: #{url}"
 
+      # For now we'll just reload ALL css to dodge pipeline stuff
+      existing_styles = Element.find("link[rel=stylesheet]")
+      existing_styles.each do | existing_style |
+        # puts "Existing style: #{existing_style.inspect}"
+        new_style = existing_style.clone
+        # puts "New style: #{new_style.inspect}"
+        existing_style.after(new_style)
+        # existing_style.remove
+        `setTimeout(function(){ existing_style.$remove() }, 300)`
```

Comment:
I don't know if this setTimeout helps, seems to work ok without it -- I copied it from https://github.com/bhauman/lein-figwheel/blob/master/support/src/figwheel/client/file_reloading.cljs#L486